src/input_map.rs

```diff
@@ -78,7 +78,7 @@ use std::marker::PhantomData;
 pub struct InputMap<A: Actionlike> {
     /// The raw vector of [PetitSet]s used to store the input mapping,
     /// indexed by the `Actionlike::id` of `A`
-    map: Vec<PetitSet<UserInput, 16>>,
+    map: Vec<PetitSet<Box<dyn InputLikeObject>, 16>>,
```

Comment:
We should just swap to a HashMap here now that we're no longer stack-allocating. Will simplify a *lot* of code.

Reply:
I assume we still want "slots" so the inputs are ordered and we can have a "secondary" input in index 1 without the "primary" index 0 being populated. Probably accomplish this with a `HashMap<usize, Box<dyn InputLikeObject>`?

Reply:
What would you think of having a type that wraps the HashMap so we can make it more clear what the purpose of the usize is? Named something `Bindings` maybe? That way when we call things like [InputMap::get](https://docs.rs/leafwing-input-manager/latest/leafwing_input_manager/input_map/struct.InputMap.html#method.get) we return our Bindings struct instead of a hashmap with a usize as key that is somewhat unclear what the purpose is.